src/containers/Widgets/Widgets.js

```diff
@@ -28,7 +28,10 @@ export default class Widgets extends Component {
   static reduxAsyncConnect(params, store) {
     const {dispatch, getState} = store;
     if (!isLoaded(getState())) {
-      return dispatch(loadWidgets());
+      const promise = dispatch(loadWidgets());
+      if (__SERVER__) {
```

Comment:
hmm weird, i thought i replied to this. Sorry for the late reply. I do think a more elegant solution would be best, not sure how other than another static method that will be called after transition or componentDidMount. Or it could even be in a form of a helper? So a callback helper to be called, then youd save having two static methods.